cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -130,6 +130,14 @@ struct ARROW_EXPORT S3Options {
   /// Whether OutputStream writes will be issued in the background, without blocking.
   bool background_writes = true;
 
+  /// Whether to allow creation or deletion of buckets
```

Comment:
It seems that "deletion" depends on `allow_create_buckets` confuses users. How about renaming `allow_create_buckets` or create one more option?

BTW, why should we disable "deletion" by default?

Reply:
We probably don't have to. Buckets can only be deleted if already empty, so hard to accidentally delete an important one I suppose.